engine/consensus/dkg/messaging_engine.go

```diff
@@ -104,40 +104,54 @@ func (e *MessagingEngine) Process(_ network.Channel, originID flow.Identifier, e
 func (e *MessagingEngine) process(originID flow.Identifier, event interface{}) error {
 	switch v := event.(type) {
 	case *msg.DKGMessage:
+		e.forwardInboundMessageAsync(originID, v)
+		return nil
+	default:
+		return fmt.Errorf("invalid event type (%T)", event)
+	}
+}
+
+func (e *MessagingEngine) forwardInboundMessageAsync(originID flow.Identifier, message *msg.DKGMessage) {
+	e.unit.Launch(func() {
 		e.tunnel.SendIn(
 			msg.PrivDKGMessageIn{
-				DKGMessage: *v,
+				DKGMessage: *message,
 				OriginID:   originID,
 			},
 		)
-		return nil
-	default:
-		return fmt.Errorf("invalid event type (%T)", event)
-	}
+	})
 }
 
 func (e *MessagingEngine) forwardOutgoingMessages() {
 	for {
 		select {
 		case msg := <-e.tunnel.MsgChOut:
-			expRetry, err := retry.NewExponential(retryMilliseconds)
-			if err != nil {
-				e.log.Fatal().Err(err).Msg("failed to create retry mechanism")
-			}
+			e.forwardOutboundMessageAsync(msg)
+		case <-e.unit.Quit():
+			return
+		}
+	}
+}
 
-			maxedExpRetry := retry.WithMaxRetries(retryMax, expRetry)
-			err = retry.Do(e.unit.Ctx(), maxedExpRetry, func(ctx context.Context) error {
-				err := e.conduit.Unicast(&msg.DKGMessage, msg.DestID)
-				if err != nil {
-					e.log.Err(err).Msg("error sending dkg message retrying")
-				}
-				return retry.RetryableError(err)
-			})
+func (e *MessagingEngine) forwardOutboundMessageAsync(message msg.PrivDKGMessageOut) {
+	f := func() {
+		expRetry, err := retry.NewExponential(retryMilliseconds)
+		if err != nil {
+			e.log.Fatal().Err(err).Msg("failed to create retry mechanism")
+		}
+
+		maxedExpRetry := retry.WithMaxRetries(retryMax, expRetry)
+		err = retry.Do(e.unit.Ctx(), maxedExpRetry, func(ctx context.Context) error {
+			err := e.conduit.Unicast(&message.DKGMessage, message.DestID)
 			if err != nil {
-				e.log.Error().Err(err).Msg("error sending dkg message")
+				e.log.Error().Err(err).Msg("error sending dkg message retrying")
 			}
-		case <-e.unit.Quit():
-			return
+			return retry.RetryableError(err)
+		})
+		if err != nil {
+			e.log.Error().Err(err).Msg("error sending dkg message")
```

Comment:
* On a first glance, this mode of error treatment seems to be conflicting with our policy 
   > We would like to avoid:
   > • swallowing errors, or "just" logging them

   (for details see Francois' great notion doc [Error handling in Flow](https://www.notion.so/dapperlabs/Error-Handling-in-Flow-a7692d60681c4063b48851c0dbc6ed10)).

* On a second glance, I think your error handling here is fine for the following reasons:
   * all we do here is handing a message to the networking layer, which we expect to error from time to time (unfortunately without specific sentinels) 
   * given that the networking layer is the _only_ source of errors, it is safe to just "swallow all errors"
 * But keep in mind that you need to untangle the code here, and inspect the detailed error propagation of `retry.Do` to arrive at this conclusion. Lastly, `retry.Do` could theoretically propagate an error from the context:
   ```golang
		select {
		case <-ctx.Done():
			return ctx.Err()
		case <-time.After(next):
			continue
		}
   ```

So in conclusion, there are multiple implicit assumptions here that all need to be satisfied in order for the code to comply with our "do not swallow errors" policy. 


#### suggestion: 
Preferably, such assumptions should be reflected by the code. Because 6 months down the line, some engineer (maybe it is you) might need to go through all of this thought process again to safely update the code. Some goDoc is still better than nothing.

Reply:
@kc1116 Could you add some documentation here explaining why swallowing the error is OK in this case, like alex suggested - what conditions can result in an error return here and why they're expected (also can note that failure to send an individual DKG message doesn't necessarily result in local or global DKG failure)